modules/nf-core/windowmasker/main.nf

```diff
@@ -0,0 +1,44 @@
+process WINDOWMASKER {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "bioconda::blast=2.13.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/blast:2.13.0--hf3cf87c_0':
+        'quay.io/biocontainers/blast:2.13.0--hf3cf87c_0' }"
+
+    input:
+    tuple val(meta), path(ref)
+
+    output:
+    tuple val(meta), path("${output}")  , emit: wm_intervals
+    path "versions.yml"                 , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args1   = task.ext.args1    ?: ""
+    def args2   = task.ext.args2    ?: ""
+    def prefix  = task.ext.prefix   ?: "${meta.id}"
+    def outfmt  = task.ext.outfmt   ?: "interval"
```

Comment:
```suggestion
    def outfmt  = task.ext.suffix   ?: "interval"
```
We have a designated suffix variable for this, but this should only be used if the output extension cannot be derived from the arguments or other inputs. So does the module determine the output format based on the name of the output file or based on an argument?

Reply:
It is based on an input argument
```
-outfmt <String, `fasta', `interval', `maskinfo_asn1_bin',
                  `maskinfo_asn1_text', `maskinfo_xml', `seqloc_asn1_bin',
                  `seqloc_asn1_text', `seqloc_xml'>
```

hence why making its own ext argument in the first place

Reply:
To keep all modules compatible we have decided to not allow this in `nf-core`. Only a very select few `task.ext.args` are allowed (see [here](https://nf-co.re/docs/contributing/modules#what-are-the-ext-propertieskeys). I suggested suffix because that one can also be used when the output format is based on the type specified by the output file.

Reply:
That makes sense. 
So in that case, the user would need to set both `ext.args = -outfmt interval` and `ext.suffix = interval`?

Reply:
In this case I think it would be best to write some code that checks which outfmt is given in ext.args and use that one as extension. That would be the most user friendly solution I think

Reply:
In that case, would something like this be ok?
```
    if (task.ext.args.contains('-outfmt')) {
        outfmt = task.ext.args.split('-outfmt')[1] // relies on -outfmt being the last arg in ext.args
    } else {
        log.info('WINDOWMASKER USTAT: outfmt not specified defaulting to "interval"')
        task.ext.args = task.ext.args + "-outfmt interval"
        outfmt = "interval"
    }
```

Reply:
Here is another way I've done it before for `bcftools`: https://github.com/nf-core/modules/blob/e73f327f6ef9f77a8e23dcf90340caa01f02954e/modules/nf-core/bcftools/convert/main.nf#L31-L35. This is indeed a bit more redundant but doesn't force the user to specifically place their outfmt at the end of args :) (yay for flexibility)

Reply:
Otherwise using a regex is also a possibility :)

Reply:
Perfect! Thanks!